eZ/Publish/Core/Helper/TranslationHelper.php

```diff
@@ -285,15 +287,34 @@ public function getAvailableLanguages()
     /**
      * @param string|null $forcedLanguage
      * @param string|null $fallbackLanguage
+     * @param string|null $mainLanguage
      *
      * @return array|mixed
      */
-    private function getLanguages($forcedLanguage = null, $fallbackLanguage = null)
+    private function getLanguages($forcedLanguage = null, $fallbackLanguage = null, $mainLanguage = null)
     {
         if ($forcedLanguage !== null) {
             $languages = array($forcedLanguage);
         } else {
             $languages = $this->configResolver->getParameter('languages');
+
+            if ($mainLanguage !== null) {
+                // Always add $mainLanguage as the first entry, because it is our main language
+                $result = array($mainLanguage);
+                foreach ($languages as $key => $lang) {
+                    if ($key === 0) {
+                        // Push the 1st prioritized language as first
+                        array_unshift($result, $lang);
+                        continue;
+                    }
+                    $result[] = $lang;
+                }
+
+                // remove duplicates
+                $languages = array_unique($result);
+
+                return $languages;
```

Comment:
```cite
Whereas "mainLanguage" must be placed in 2nd position in the returned array, because it has to be the chosen language if the 1st prioritorized language translation is not available.
```
ok... but how will it play with that old setting "ShowUntraslatedObjects"? I mean, if you don't want your contents be shown if there is no translation for them for the selection siteaccess...